include/dfu/pcd.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/** @file pcd.h
+ *
+ * @defgroup pcd Peripheral Core DFU
+ * @{
+ * @brief API for handling DFU of peripheral cores.
+ *
+ * The PCD API provides functions for sending/receiving DFU images between
+ * cores on a multi core system. The cores communicate through a command
+ * structure (CMD) which is stored in a shared memory location.
+ *
+ */
+
+#ifndef PCD_H__
+#define PCD_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <device.h>
+
+#ifdef CONFIG_SOC_SERIES_NRF53X
+
+/* These must be hard coded as this code is preprocessed for both net and app
+ * core.
+ */
+#define APP_CORE_SRAM_START 0x20000000
+#define APP_CORE_SRAM_SIZE KB(512)
+#define RAM_SECURE_ATTRIBUTION_REGION_SIZE 0x2000
+#define PCD_CMD_ADDRESS (APP_CORE_SRAM_START \
+			+ APP_CORE_SRAM_SIZE \
+			- RAM_SECURE_ATTRIBUTION_REGION_SIZE)
+#endif
+
+
+/** @brief Opaque type */
+struct pcd_cmd;
+
+/** @brief Get a PCD CMD from the specified address.
+ *
+ * @param addr The address to check for a valid PCD CMD.
+ *
+ * @retval A pointer to the PCD CMD if successful.
+ *           Otherwise, NULL is returned.
+ */
+struct pcd_cmd *pcd_cmd_get(void *addr);
+
+/** @brief Construct a PCD CMD for copying data/firmware.
+ *
+ * @param cmd_addr The address to write the CMD to.
+ * @param src_addr The address to which the CMD copy data from.
+ * @param len      The number of bytes that should be copied.
+ * @param offset   The offset within the flash device to write the data to.
+ *
+ * @retval A pointer to the written PCD CMD if successful.
+ *           Otherwise, NULL is returned.
+ */
+struct pcd_cmd *pcd_cmd_write(void *cmd_addr, const void *src_addr, size_t len,
+		  size_t offset);
+
+/** @brief Invalidate the PCD CMD, indicating that the copy failed.
+ *
+ * @param cmd The PCD CMD to invalidate.
+ *
+ * @retval non-negative integer on success, negative errno code on failure.
+ */
+int pcd_invalidate(struct pcd_cmd *cmd);
```

Comment:
If this function is invalidating a command, you can infix it with `_cmd_` like other functions in the API that work on `struct pcd_cmd`.